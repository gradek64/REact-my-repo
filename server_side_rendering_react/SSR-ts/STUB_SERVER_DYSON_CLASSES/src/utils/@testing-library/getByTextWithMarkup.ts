import { screen } from '@testing-library/react'

const matcher = (text) => {
  return (_, node: HTMLElement) => {
    const hasText = (innerNode: HTMLElement) => innerNode.textContent === text
    const childrenDontHaveText = Array.from(node.children).every((child: HTMLElement) => !hasText(child))
    return hasText(node) && childrenDontHaveText
  }
}

export const getByTextWithMarkup = (text: string): ReturnType<(typeof screen)['getByText']> =>
  screen.getByText(matcher(text))

export const queryByTextWithMarkup = (text: string): ReturnType<(typeof screen)['queryByText']> =>
  screen.queryByText(matcher(text))
