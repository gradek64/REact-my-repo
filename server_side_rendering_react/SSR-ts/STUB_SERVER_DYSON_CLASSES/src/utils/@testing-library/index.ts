export * from './renderer'
export { waitForAnimatedSubmitButton } from './waitForAnimatedSubmitButton'
export { getByTextWithMarkup, queryByTextWithMarkup } from './getByTextWithMarkup'

// eslint-disable-next-line import/no-extraneous-dependencies
export { default as userEvent } from '@testing-library/user-event'

export { default as creditPlanSelectors } from './selectors/creditPlans'
export { renderWithModalWrapper } from './renderWithModalWrapper'
