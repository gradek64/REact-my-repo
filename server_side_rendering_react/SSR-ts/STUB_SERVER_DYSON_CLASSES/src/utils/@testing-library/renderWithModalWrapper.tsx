import React, { ReactNode } from 'react'
import { Modal } from '@sainsburys-tech/bolt'
import { RenderResult, render } from '.'

export const renderWithModalWrapper = (children: ReactNode, renderOptions = {}, modalProps = {}): RenderResult =>
  render(
    <Modal show {...modalProps}>
      {children}
    </Modal>,
    renderOptions,
  )
