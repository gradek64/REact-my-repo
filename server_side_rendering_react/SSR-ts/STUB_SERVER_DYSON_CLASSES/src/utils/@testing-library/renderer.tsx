/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React, { FC, ReactElement } from 'react'
// eslint-disable-next-line no-restricted-imports
import { render, configure, RenderResult } from '@testing-library/react'

import { Provider } from 'react-redux'
import { createBrowserHistory } from 'history'
import { ConnectedRouter } from 'connected-react-router'
import { ThemeProvider } from 'styled-components'
import { Themes } from '@sainsburys-tech/bolt'
import { I18nextProvider } from 'react-i18next'
import { PreloadedState } from 'redux'

import i18n from '../../../config/i18n/test'
import rootSaga from '../../sagas'
import createStoreActual from '../../store'
import { RootState } from '../../types/reducers'
import { initialState as reducerInitialState } from '../../config/testUtils/__mocks__/state'

const browserHistory = createBrowserHistory()

configure({ testIdAttribute: 'data-test' })

export const createStore = (
  initialState: PreloadedState<RootState>,
  initialHistory: typeof browserHistory | undefined,
): ReturnType<typeof createStoreActual> => {
  const history = initialHistory || browserHistory
  const store = createStoreActual(history, initialState)

  store.runSaga(rootSaga)
  return store
}

/* https://testing-library.com/docs/react-testing-library/setup */
const customRender = (
  ui: ReactElement,
  {
    initialState = reducerInitialState,
    initialHistory = browserHistory,
    store = createStore(initialState, initialHistory),
    ...renderOptions
  } = {},
): RenderResult => {
  const AllTheProviders: FC = ({ children }) => {
    return (
      <I18nextProvider i18n={i18n}>
        <Provider store={store}>
          <ConnectedRouter history={initialHistory}>
            <ThemeProvider theme={Themes.ArgosTheme}>{children}</ThemeProvider>
          </ConnectedRouter>
        </Provider>
      </I18nextProvider>
    )
  }
  return render(ui, { wrapper: AllTheProviders, ...renderOptions })
}

// eslint-disable-next-line no-restricted-imports
export * from '@testing-library/react'

export { customRender as render }
