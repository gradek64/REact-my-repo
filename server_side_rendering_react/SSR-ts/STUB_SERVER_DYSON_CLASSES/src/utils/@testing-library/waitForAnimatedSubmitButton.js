import { screen, fireEvent, waitFor, act } from '@testing-library/react'

/**
 * Helper method to auto click and assert values on an <AnimatedSubmitButton> or <AnimatedSubmitButtonWithRetries> component
 * @param {object.labelSubmit} string - (required) Initial text the button displays
 * @param {object.labelSubmitting} string - (optional) Submitting text state the button displays whilst submitting (leave undefined to skip this assertion)
 * @param {object.labelSubmitted} string - (optional) Submitted text state (leave undefined to skip this assertion)
 * @param {object.spinnerTestId} string - (required) `data-test` attribute for the spinner <circle> element
 * @returns {promise}
 */

export async function waitForAnimatedSubmitButton({ labelSubmit, labelSubmitting, labelSubmitted, spinnerTestId }) {
  const waitForOptions = { timeout: 5000 }
  await expect(screen.getByText(labelSubmit)).toBeInTheDocument()

  fireEvent.click(screen.getByText(labelSubmit))

  if (labelSubmitting) {
    await waitFor(() => {
      expect(screen.getByText(labelSubmitting)).toBeInTheDocument()
    }, waitForOptions)
  }

  /* Wait for the spinner to show, then manually trigger a spin iteration on
    it due to css events not firing in JSDOM */
  jest.useFakeTimers()

  await waitFor(() => {
    const spinner = screen.queryByTestId(spinnerTestId)
    if (spinner) {
      fireEvent.animationIteration(spinner)
    }
  }, waitForOptions)

  act(() => {
    jest.runAllTimers()
  })

  if (labelSubmitted) {
    await waitFor(() => {
      expect(screen.getByText(labelSubmitted)).toBeInTheDocument()
    }, waitForOptions)
  }

  jest.useRealTimers()
}
