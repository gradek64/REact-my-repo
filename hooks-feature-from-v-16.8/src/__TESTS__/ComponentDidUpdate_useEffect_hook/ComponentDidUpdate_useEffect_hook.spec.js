import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CounterComp from '../../ComponentDidUpdate_useEffect_hook';

describe('should render correclty with initial state', () => {
  afterEach(cleanup);
  it('should take a snapshot', () => {
    const { asFragment } = render(<CounterComp />);
    expect(asFragment(<CounterComp />)).toMatchSnapshot();
  });
});

describe('state response to clicks', () => {
  afterEach(cleanup);
  it('increments counter', () => {
    //use container cheaky way to have access to standard DOM selectors
    const { container } = render(<CounterComp />);
    const button1 = container.querySelectorAll('button')[0];
    fireEvent.click(button1);
    expect(container.querySelector('#display')).toHaveTextContent(
      'You clicked 1 times'
    );
  });
  it('runs and updates useEffect [count] state property', () => {
    //use container cheaky way to have access to standard DOM selectors
    const { container } = render(<CounterComp />);
    const button1 = container.querySelectorAll('button')[0];
    fireEvent.click(button1);
    expect(document.title).toBe('updated count state:1');
  });
});
