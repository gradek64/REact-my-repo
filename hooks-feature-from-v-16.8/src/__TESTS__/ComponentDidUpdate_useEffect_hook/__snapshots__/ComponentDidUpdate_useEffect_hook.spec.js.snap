// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correclty with initial state should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Simple click update counter using hook useState()
    </h2>
    <p
      data-testid="counter"
      id="display"
    >
      You clicked 0 times
    </p>
    <button
      type="button"
    >
      Click me up
    </button>
  </div>
</DocumentFragment>
`;
